### PR TITLE
Use in-memory token store when developing locally

### PR DESCRIPTION
**Pull request by marcus-bcl**
Removes the need for a local Redis container

### CODE REVIEW

File: server/data/index.ts [hunk 1]
```diff
@@ -13,13 +13,17 @@ buildAppInsightsClient(applicationInfo)
...
+import TestingTokenStore from './tokenStore/testingTokenStore'
+import config from '../config'
 
 type RestClientBuilder<T> = (token: string) => T
 
 export const dataAccess = () => ({
   applicationInfo,
-  hmppsAuthClient: new HmppsAuthClient(new TokenStore(createRedisClient())),
+  hmppsAuthClient: new HmppsAuthClient(
+    config.redis.enabled ? new RedisTokenStore(createRedisClient()) : new TestingTokenStore(),
```
Comment (andrewrlee) [inline server/data/index.ts:25]: Worth calling this InMemoryToken store? 
We use an in memory token store for jobs as it means don't need to worry about redis being accessible (dodgy DNS issues)

File: server/middleware/setUpWebSession.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 import RedisStore from 'connect-redis'
 import express, { Router } from 'express'
 import { createRedisClient } from '../data/redisClient'
 import config from '../config'
 import logger from '../../logger'
 
 export default function setUpWebSession(): Router {
-  const client = createRedisClient()
-  client.connect().catch((err: Error) => logger.error(`Error connecting to Redis`, err))
+  let store: Store
```
Comment (andrewrlee) [inline server/middleware/setUpWebSession.ts:10]: should this fall back to the inmemory store?  what happens if you pass undefined?

File: server/data/index.ts (hunk 1 above)
Comment (marcus-bcl) [reply server/data/index.ts:25]: Good call, I'll rename it

File: server/middleware/setUpWebSession.ts (hunk 2 above)
Comment (marcus-bcl) [reply server/middleware/setUpWebSession.ts:10]: Express defaults to a MemoryStore, so passing undefined means it uses its own in-memory store for session data.

**Pull request merged**